### PR TITLE
Feature/data filtering

### PR DESCRIPTION
**Pull request by martinbagshaw**
- A work in progress, initial filter functions to filter search results from Jamie's API. No tests yet, but pushed in case I get hit by the proverbial bus.
- Uses a node module to use fetch in node (as opposed to the front end). This may not be required in the final version.
- Relates to #2 and #3

### CODE REVIEW

**Comment by martinbagshaw** [comment]
For viewers: I do not intend for this branch to be merged, more used as a point of reference for data filtering functions we might use or adapt to filter API results with.

I will probably continue working on this sort of thing unless someone says I shouldn't - been enjoying it, and I feel it may be useful.

**Comment by martinbagshaw** [comment]
@thejoefriel I have hit a bit of a blocker here, I was wondering if you could help me out?

When running deepEquals tests, it appears that some test data I am importing is getting mutated. If I comment out either one of the last tests in `filterFunctions.test.js`, tests all pass.

I was wondering if you have encountered this before, and if so, how you solved it?

Thanks.

File: helpers/filterByCountry.js [hunk 1]
```diff
@@ -2,43 +2,72 @@
...
+	status = null
+) => {
+	const resultArray = resultObject.results;
+	// const keys = Object.keys(resultArray);
+
+	// we have to clone the array because we are overriding
+	// the Locations reference in line 45
+	// this one> item.Locations = filteredByCountry;
+	// this is easy or we can use recursion
+	const clonedArr = JSON.parse(JSON.stringify(resultArray));
```
Comment (martinbagshaw) [inline helpers/filterByCountry.js:28]: I like this one, short and sweet

**Review by martinbagshaw — COMMENTED** [review]
Looks good to me. To elaborate on what I said on slack, I think it may make sense to make a **utils** or **dataFunctions** folder in the latest master, and bring in some of these functions and tests to their own files.

Regarding git diffs, I find linting code can make it quite hard to reason about what has actually changed. For example here, you have only changed one line, but large sections are green or red.

An alternative to this could be to make the changes without linting and commit, then make a commit that **just lints files**, with the commit message **lint**. This will make it easy for others to tell the difference between code changes and formatting changes, as they will view the commit without the **lint** commit message.

**Comment by MohammedYehia** [comment]
@mr-bagglesworth  sorry for that that was the prettier thing I forgot to stop it.
I agree and I will try and do some test as we agreed

**Pull request closed without merging**